src/bp-core/bp-core-functions.php

```diff
@@ -4514,6 +4514,26 @@ function bp_find_mentions_by_at_sign( $mentioned_users, $content ) {
 	$pattern = '/(?<=[^A-Za-z0-9]|^)@([A-Za-z0-9-_\.@]+)\b/';
 	preg_match_all( $pattern, $content, $usernames );
 
+	// Get urls from content that contains @ sign.
+	preg_match_all( '#\bhttps?://[^\s]+@[^\s]+#', $content, $usernames_with_url );
+
+	// Remove the username which is part of url.
+	if ( ! empty( $usernames_with_url[0] ) && ! empty( $usernames[1] ) ) {
+		foreach ( $usernames_with_url[0] as $url ) {
+			// Remove tags if any in url.
+			$url = strip_tags( $url );
+
+			// Get username from url.
+			$username = explode( '@', $url );
+			$username = end( $username );
+			$key      = array_search( $username, $usernames[1] );
+			if ( false !== $key ) {
+				unset( $usernames[1][ $key ] );
+				unset( $usernames[0][ $key ] );
+			}
+		}
+	}
```

Comment:
Instead of unset here, please try to update the original pattern which extract the mention users.